numpy/core/code_generators/ufunc_docstrings.py

```diff
@@ -1835,6 +1835,17 @@ def add_newdoc(place, name, doc):
     >>> np.left_shift(5, [1,2,3])
     array([10, 20, 40])
 
+    Note that the dtype of the second argument may change the dtype of the
+    result and can lead to unexpected results in some cases (see
+    :ref:`Casting Rules <ufuncs.casting>`):
+
+    >>> np.left_shift(np.array([255], dtype='u1'), np.array([1], dtype='u1'))
+    array([254], dtype=uint8)
+    >>> np.left_shift(np.array([255], dtype='u1'), np.array([1], dtype='i1'))
+    array([510], dtype=int16)
+    >>> np.left_shift(np.uint8(255), 1)
```

Comment:
Perhaps you want to lead with the scalar example? It might be helpful to have a comment that gives the expected result (254) .

Reply:
Yes, those examples were not conveying a very clear picture.  It should be clearer now.  I removed the array examples to keep it simple and clear.